src/operator/make_loss-inl.h

```diff
@@ -0,0 +1,134 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file make_loss-inl.h
+ * \brief special layer for propagating loss
+*/
+#ifndef MXNET_OPERATOR_MAKE_LOSS_INL_H_
+#define MXNET_OPERATOR_MAKE_LOSS_INL_H_
+#include <dmlc/logging.h>
+#include <dmlc/parameter.h>
+#include <mxnet/operator.h>
+#include <cstring>
+#include <map>
+#include <string>
+#include <vector>
+#include <utility>
+#include "./mshadow_op.h"
+#include "./operator_common.h"
+
+namespace mxnet {
+namespace op {
+
+namespace make_loss_enum {
+enum MakeLossOpInputs {kData};
+enum MakeLossOpOutputs {kOut};
+}  // namespace make_loss_enum
+
+struct MakeLossParam : public dmlc::Parameter<MakeLossParam> {
+  float grad_scale;
+  DMLC_DECLARE_PARAMETER(MakeLossParam) {
+    DMLC_DECLARE_FIELD(grad_scale).set_default(1.0f)
+    .describe("gradient scale as a supplement to unary and binary operators");
+  }
+};
+
+template<typename xpu>
+class MakeLossOp : public Operator {
+ public:
+  explicit MakeLossOp(MakeLossParam param) : param_(param) {}
+
+  virtual void Forward(const OpContext &ctx,
+                        const std::vector<TBlob> &in_data,
+                        const std::vector<OpReqType> &req,
+                        const std::vector<TBlob> &out_data,
+                        const std::vector<TBlob> &aux_args) {
+    using namespace mshadow;
+    using namespace mshadow::expr;
+    CHECK_EQ(in_data.size(), 1) << "MakeLoss can only be used to one input";
+    CHECK_EQ(out_data.size(), 1);
+    CHECK_EQ(in_data[make_loss_enum::kData].ndim(), 2)
+    << "MakeLoss applies to all unary and binary operator with 2 dimension input";
+    Stream<xpu> *s = ctx.get_stream<xpu>();
+    Tensor<xpu, 2> data = in_data[make_loss_enum::kData].get<xpu, 2, real_t>(s);
+    Tensor<xpu, 2> out = out_data[make_loss_enum::kOut].get<xpu, 2, real_t>(s);
+    Assign(out, req[make_loss_enum::kData], F<mshadow_op::identity>(data));
```

Comment:
You can drop the F<mshadow_op::identity>.
You can even save a copy if req[.] is WriteInplace which should be the case most of the times.
It's not a big deal though.